src/webportal/src/app/job-submission/utils/utils.js

```diff
@@ -134,16 +140,47 @@ function addPreCommandsToProtocolTaskRoles(protocol, preCommands) {
   });
 }
 
+function addPostCommandsToProtocolTaskRoles(protocol) {
```

Comment:
Why the function name is `addPostCommandsToProtocolTaskRoles`? I think tensor board command is execute before user command.

Rename it to `addTensorBoardCommandsToProtocolTaskRoles` and call it in `addPreCommandsToProtocolTaskRoles`?

Reply:
Tensorboard is not a background process. It will always be in the foreground when the it is running. So it will block the execution of the user script. Maybe it is possible to run TensorBoard in the background, I need to test it further.

Reply:
Thanks, it make sense. Add comments for this?

Reply:
Thanks. Comments added.